app/util/analyticsUtils.js

```diff
@@ -36,8 +36,9 @@ export function addAnalyticsEvent(event) {
  * @return string
  */
 export function getAnalyticsInitCode(config, hostname) {
-  if (config.analyticsScript && hostname && !hostname.match(/dev|test/)) {
-    return config.analyticsScript(hostname);
+  const host = hostname?.match(/dev|test/) ? config.devAnalyticsHost : hostname;
+  if (config.analyticsScript && host) {
```

Comment:
```suggestion
  if (host && config.analyticsScript) {
```
Fail fast in case `host` ends up `undefined`